solidity/contracts/maintainer/MaintainerProxy.sol

```diff
@@ -0,0 +1,373 @@
+///         the spent gas from the Reimbursement Pool. Only the authorized
+///         maintainers can proxy call Bridge contract.
+contract MaintainerProxy is Ownable, Reimbursable {
+    Bridge public bridge;
+
+    /// @notice Authorized maintainer that can interact with the maintainer proxy
+    ///         contract. Authorization can be granted and removed by the governance.
+    mapping(address => bool) public isAuthorized;
+
+    /// @notice Gas that is meant to balance the request of a new wallet overall
+    ///         cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _requestNewWalletGasOffset;
+
+    /// @notice Gas that is meant to balance the submission of deposit sweep proof
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _submitDepositSweepProofGasOffset;
+
+    /// @notice Gas that is meant to balance the submission of redemption proof
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _submitRedemptionProofGasOffset;
+
+    /// @notice Gas that is meant to balance the notification of closeable wallet
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _notifyCloseableWalletGasOffset;
+
+    /// @notice Gas that is meant to balance the defeat fraud challenge
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _defeatFraudChallengeGasOffset;
+
+    /// @notice Gas that is meant to balance the submission of moving funds proof
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _submitMovingFundsProofGasOffset;
+
+    /// @notice Gas that is meant to balance the submission of moving funds commitment
+    ///         overall cost. Can be updated by the governance based on the current
+    ///         market conditions.
+    uint256 internal _submitMovingFundsCommitmentGasOffset;
+
+    event MaintainerAuthorized(address indexed maintainer);
+
+    event MaintainerUnauthorized(address indexed maintainer);
+
+    event BridgeUpdated(address newBridge);
+
+    event GasOffsetParametersUpdated(
+        uint256 requestNewWalletGasOffset,
+        uint256 submitDepositSweepProofGasOffset,
+        uint256 submitRedemptionProofGasOffset,
+        uint256 notifyCloseableWalletGasOffset,
+        uint256 defeatFraudChallengeGasOffset,
+        uint256 submitMovingFundsProofGasOffset,
+        uint256 submitMovingFundsCommitmentGasOffset
+    );
+
+    modifier onlyMaintainer() {
+        require(isAuthorized[msg.sender], "Caller is not authorized");
+        _;
+    }
+
```

Comment:
Why don't we override [`onlyReimbursableAdmin`](https://github.com/keep-network/keep-core/blob/222be833fd41dac7255974aa885693cb278ecdd2/solidity/random-beacon/contracts/Reimbursable.sol#L38-L40)? 

It would be great to unit test that `updateReimbursementPool` is possible.

Reply:
Good catch, but the compiler doesn't enforce overriding this modifier, which IMO is a bit bigger issue than what we have in `Reimbursable`. Instead of what we have in [here](https://github.com/keep-network/keep-core/blob/main/solidity/random-beacon/contracts/Reimbursable.sol#L38) I think should be changed to just `modifier onlyReimbursableAdmin() virtual` which will enforce implementation of the `onlyReimbursableAdmin` modifier. I've tried that but the `solhint` has an unresolved [bug](https://github.com/protofire/solhint/issues/281) and the linter complains about virtual modifiers. There is a workaround in the comments under that bug but it will require more work.

Reply:
Tests for the `updateReimbursementPool` were also added